pkg/util/sortedset/sortedset.go

```diff
@@ -0,0 +1,182 @@
+package sortedset
```

Comment:
this should be called something like rankedset, a soretedset is a different thing
Also prese move this whole commit to a different PR, as it has nothing to do with the rest of infrastrcture changes